src/components/eip3643/User.tsx

```diff
@@ -1,19 +1,53 @@
 import { useEffect, useContext, useState } from "react";
 import TransferToken from "@/components/eip3643/user/TransferToken";
 import { Eip3643Context } from "@/contexts/Eip3643Context";
-import { readTokenName } from "@/services/contracts/wagmiGenActions";
+import {
+  readTokenName,
+  readTokenIdentityRegistry,
+  watchIdentityRegistryIdentityRegisteredEvent,
+} from "@/services/contracts/wagmiGenActions";
 import { TokenNameItem } from "@/types/types";
 import { useWalletInterface } from "@/services/wallets/useWalletInterface";
 import { Divider, Select, Stack } from "@chakra-ui/react";
+import { WatchContractEventReturnType } from "@/services/contracts/watchContractEvent";
 
 export default function User() {
   const [tokenSelected, setTokenSelected] = useState(
     null as TokenNameItem | null,
   );
   const [tokens, setTokens] = useState([] as Array<TokenNameItem>);
+  const [registeredIdentities, setRegisteredIdentities] = useState<string[]>();
   const { accountEvm } = useWalletInterface();
   const { deployedTokens } = useContext(Eip3643Context);
 
+  useEffect(() => {
```

Comment:
I'd suggest to make a reusable hook with this part, considering we have the same watcher in the `RegisterIdentity` component for DRY reasons.